### ISSUE TITLE
Ensure that Dimension parameters are consistently respected

### ISSUE THREAD

**Issue by philippjfr** [ioam/holoviews#2468]
``Dimension`` objects in HoloViews define a wide range of parameters which are not always consistently respected. I therefore wanted to open a meta-issue to collect various examples where we could adhere to the dimensions more consistently.

## Dimension.type

- [ ] Should define the type of the axis, i.e. numeric, categorical or datetime (https://github.com/ioam/holoviews/issues/1312)
- [ ] Should allow defining the type of widget (eventually with more flexibility after the widget refactor)

## Dimension.values

- [ ] Use as explicit order when plotting GridSpace
- [ ] Use as categories on categorical axis

**Comment by jlstevens** [comment]
I know we want to get rid of ``extents`` and have plotting ranges (e.g ``xlim``, ``ylim`` etc) but will that override the dimension ranges? I know there is ``soft_range`` but I find that is very rarely used in practice...

**Comment by philippjfr** [comment]
>I know we want to get rid of extents and have plotting ranges (e.g xlim, ylim etc) but will that override the dimension ranges?

Yes explicitly set ranges at the plot level should have highest precedence just like extents does now.

>I know there is soft_range but I find that is very rarely used in practice...

I don't think that's an issue, it's used an consistently respected, my mine worry is about Dimension parameters which are only partially respected.
